provider/provider.go

```diff
@@ -1339,10 +1350,22 @@ func setFieldViaOp(typ string, attrs map[string]interface{}, name string, key st
 	var diags diag.Diagnostics
 
 	valStr := attrValueString(typ, key, val, attrs)
-	appendActionLog(fmt.Sprintf("Setting %s field: '%s'.'%s' :: %+v\n", typ, name, key, val))
-
 	op := fmt.Sprintf("%s.%s = %s", name, key, valStr)
 
+	if typ == "dashboard" {
+		isPrimary := GetNestedValueOrDefault(attrs, ToKeyPath(key+".primary"), false).(bool)
+		if isPrimary {
+			appendActionLog(fmt.Sprintf("Skipping setting %s field %s...\n", typ, key))
+			return nil
+		} else {
+			if key == "groups" || key == "values" {
+				op = fmt.Sprintf("%s.%s = %s", name, key, val)
+			}
+		}
```

Comment:
Why do we skip primary here? Other symbols have a primary field that can be updated.